machine_learning/xptracking_catboost/code.py

```diff
@@ -0,0 +1,82 @@
+import dataiku
+import mlflow
+import mlflow.catboost
+import pandas as pd
+from catboost import CatBoostClassifier, Pool, cv
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelBinarizer
+from sklearn import metrics
+from dataikuapi.dss.ml import DSSPredictionMLTaskSettings
+
+# !! - Replace these values by your own - !!
+USER_PROJECT_KEY = "YOUR-PROJECT-KEY"
+USER_XPTRACKING_FOLDER_ID = "YOUR-XP-TRACKING-FOLDER-ID"
+USER_EXPERIMENT_NAME = "YOUR-EXPERIMENT-NAME"
+USER_TRAINING_DATASET = "YOUR-TRAINING-DATASET-NAME"
+USER_MLFLOW_CODE_ENV_NAME = "YOUR-MLFLOW-CODE-ENV-NAME"
+
+client = dataiku.api_client()
+project = client.get_project(USER_PROJECT_KEY)
+
+# (1)
+ds = dataiku.Dataset(USER_TRAINING_DATASET)
+df = ds.get_dataframe()
+
+cat_features= ["job", "marital", "education", "default", "housing","loan", "month", "contact", "poutcome"]
+num_features= ["age", "balance", "day", "duration", "campaign"]
+target_col ="y"
+X = df.drop(target_col, axis=1)
+y = LabelBinarizer().fit_transform(df[target_col[0]])
+
+# (2)
+params = {
+    'iterations': 100,
+    'learning_rate': 0.1, 
+    'depth': 10,
+    'cat_features': cat_features,
+    'loss_function': 'Logloss',
+    'eval_metric': 'AUC',
+    'early_stopping_rounds': 5,
+    'use_best_model': True,
+    'random_seed': 42,
+}
+
+# (3)
+mlflow_extension = project.get_mlflow_extension()
+with project.setup_mlflow(managed_folder=USER_XPTRACKING_FOLDER_ID) as mlflow:
+    mlflow.set_experiment(experiment_name=USER_EXPERIMENT_NAME)
+    with mlflow.start_run() as run:
+        run_id = run.info.run_id
+        
+        cv_dataset = Pool(
+            data=X, label=y, cat_features= cat_features)
+
+        scores = cv(cv_dataset,
+                    params,
+                    fold_count=5,
+                    seed=42,
+                    plot= False)
+        
+        for x in range(len(scores.index)):
+            mlflow.log_metric(key='mean_AUC', value=scores['test-AUC-mean'][x], step=x+1)
```

Comment:
change to `step=x` since it's fine to start at 0 & looks better too
![step_equals_zero](https://user-images.githubusercontent.com/10568658/182607458-52ad6992-824e-4715-8c69-ccf1e939adfd.png)